l1-contracts/src/core/messagebridge/Outbox.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {MessageBox} from "./MessageBox.sol";
+
+/**
+ * @title Outbox
+ * @author Aztec Labs
+ * @notice Lives on L1 and is used to consume L2 -> L1 messages. Messages are inserted by the rollup contract
+ * and will be consumed by the portal contracts.
+ */
+contract Outbox is MessageBox {
+  error Outbox__Unauthorized();
+  error Outbox__WrongChainId();
+
+  /**
+   * @dev  struct for sending messages from L2 to L1
+   * @param sender - The sender of the message
+   * @param recipient - The recipient of the message
+   * @param content - The content of the message (application specific) padded to bytes32 or hashed if larger.
+   */
+  struct L2ToL1Msg {
+    L2Actor sender;
+    L1Actor recipient;
+    bytes32 content;
+  }
+
+  // to make it easier for portal to know when to consume the message.
+  event MessageAdded(bytes32 indexed entryKey);
+
+  event MessageConsumed(bytes32 indexed entryKey, address indexed recipient);
+
+  constructor(address _registry) MessageBox(_registry) {}
+
+  /**
+   * @notice Computes an entry key for the Outbox
+   * @param _message - The L2 to L1 message
+   * @return The key of the entry in the set
+   */
+  function computeEntryKey(L2ToL1Msg memory _message) public pure returns (bytes32) {
+    // FIXME: Replace mod P later on when we have a better idea of how to handle Fields.
```

Comment:
Use `TODO` or `@todo` instead of `FIXME` to keep it more consistent across codebase.